internal/dockercompose/compose.go

```diff
@@ -2,55 +2,150 @@ package dockercompose
 
 import (
 	"context"
-	"errors"
 	"fmt"
+	"io/ioutil"
 	"os/exec"
+	"path"
 	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/windmilleng/tilt/internal/dockerfile"
+	"github.com/windmilleng/tilt/internal/model"
+
+	"gopkg.in/yaml.v2"
 )
 
+// Go representations of docker-compose.yml
+// (Add fields as we need to support more things)
+type Config struct {
+	Services map[string]ServiceConfig `yaml:"services"`
+}
+
+type ServiceConfig struct {
+	Build BuildConfig `yaml:"build"`
+}
+
+type BuildConfig struct {
+	Context    string `yaml:"context"`
+	Dockerfile string `yaml:"dockerfile"`
+}
+
+// A docker-compose service, according to Tilt.
 type Service struct {
-	Name string
+	Name    string
+	Context string
+	DfPath  string
 }
 
-func ParseConfig(ctx context.Context, files []string) ([]Service, []string, error) {
-	var args []string
-	for _, f := range files {
-		args = append(args, "-f", f)
+func (c Config) GetService(name string) (Service, error) {
+	svcConfig, ok := c.Services[name]
+	if !ok {
+		return Service{}, fmt.Errorf("no service %s found in config", name)
 	}
-	args = append(args, "config")
-	_, err := dcOutput(ctx, args...)
-	if err != nil {
-		return nil, files, err
+
+	df := svcConfig.Build.Dockerfile
+	if df == "" && svcConfig.Build.Context != "" {
+		// We only expect a Dockerfile if there's a build context specified.
+		df = "Dockerfile"
 	}
 
-	args = append(args, "--services")
-	servicesText, err := dcOutput(ctx, args...)
+	return Service{
+		Name:    name,
+		Context: svcConfig.Build.Context,
+		DfPath:  path.Join(svcConfig.Build.Context, df),
+	}, nil
+}
+
+func svcNames(ctx context.Context, configPath string) ([]string, error) {
+	servicesText, err := dcOutput(ctx, configPath, "config", "--services")
 	if err != nil {
-		return nil, files, err
+		return nil, err
 	}
 
 	serviceNames := strings.Split(string(servicesText), "\n")
 
-	var services []Service
+	var result []string
 
 	for _, name := range serviceNames {
 		if name == "" {
 			continue
 		}
-		services = append(services, Service{Name: name})
+		result = append(result, name)
+	}
+
+	return result, nil
+}
+
+func ParseConfig(ctx context.Context, configPath string) ([]Service, error) {
+	configOut, err := dcOutput(ctx, configPath, "config")
+	if err != nil {
+		return nil, err
+	}
+
+	config := Config{}
+	err = yaml.Unmarshal([]byte(configOut), &config)
+	if err != nil {
+		return nil, err
+	}
+
+	svcNames, err := svcNames(ctx, configPath)
+	if err != nil {
+		return nil, err
+	}
+
+	var services []Service
+
+	for _, name := range svcNames {
+		if name == "" {
```

Comment:
i think you already filtered these out in the svcNames function?

Reply:
ah yes, copy 🍝